\subsection{PCM to PWM conversion algorithm}
Matrix of positional counts (PCM) can be transformed to PWM using the following formula [Lifanov2003]:
\begin{equation}PWM_{\alpha,j} = \ln\frac{ PCM_{\alpha,j} + aq_{\alpha} }{ (W+a)q_{\alpha} }\,, \end{equation}

where $\alpha$~is a nucleotide (or dinucleotide) index and $j$~is a position index; $W$~is the total weight of the alignment (or the number of aligned words), $a$~is the
pseudocount value, and $q_{\alpha}$~is the background probability of nucleotide letter~$\alpha$.

Pseudocount is taken by default as the $\ln{W}$ but can be explicitly specified by user.

Alignment weight $W$ is typically a total number of aligned words and can be calculated from
 a given PCM as a sum of nucleotide counts in a particular column : $W_i = \sum_{\alpha}PCM_{\alpha,i}$.
 $W_i$~is the alignment weight for \mbox{$i$-th} position.
 Typically each position has the same alignment weight~$W$, but multiple local alignment algorithms may
 produce positional count matrices with different weights~$W_i$ of words covering each position
 (e.g. flanks can have less weight than a central part of motif). Thus the weight is safer to calculate separately for each motif position.

For \PcmToPwm\ conversion *-APE tools use a slightly modified formula:
\begin{equation}PWM_{\alpha,j} = \ln\frac{ PCM_{\alpha,j} + a_j q_{\alpha} }{ (W_j+a_j)q_{\alpha} }\end{equation}

Here $a_j$ is a pseudocount related to \mbox{$j$-th} position.
It can be either fixed for each position or equal to a logarithm of corresponding alignment weight: $a_j=\ln{W_j}$

By default all tools accept weight matrices (i.e. already converted using any similar procedure).
