\subsubsection{Obtaining PWM from PCM and PPM models}
To load motif from position count matrices there is a special \cmdoption{--pcm} option.
A similar option \cmdoption{--ppm} words for positional probability matrices (see fig.~\ref{motif-conversion-types}).

The \PcmToPwm\ or \PpmToPwm\ data model transformations can be configured.

The \PcmToPwm\ conversion is described in a \ref{Pcm2PwmMath} section. It's possible to manually specify a fixed pseudocount $a$ with \cmdoption{--pseudocount \requiredarg{a}} option. When not specified, pseudocount is derived from alignment weight $W$: $a =\ln(W)$.

\PpmToPwm\ conversion is done in two stages. At first PPM is multiplied by a constant alignment weight $W$ to obtain a PCM. Then this PCM is converted to a PWM as described above.
For the \PpmToPwm conversion, a user should supply alignment weight $W$ (for example it can be the total count of words in the initial alignment) explicitly by the \cmdoption{--effective-count \requiredarg{W}} option. If this information is not given, alignment weight of 100.0 will be used as a default assumption.

\PcmToPwm\ conversion will take the user-specified background into account.

DiPCMs are converted to diPWMs using the same formula as for \PcmToPwm\ conversion, the only difference is that now nucleotide index goes through 16 dinucleotides at each position instead of 4 nucleotides.

Possible configuration options can be seen on a fig.~\ref{motif-conversion-configuration}.
