\cmdname{SNPScan} uses a list of sequences with SNVs as input data.

The list of sequences with SNVs should be given in a single plain text file. 
Each sequence should be presented at a separate line using the following format:\\
\texttt{\inangles{SNV name} \inangles{left flank}[\inangles{variant 1}/\inangles{variant 2}]\inangles{right flank}}

SNV name shouldn't contain empty delimiters (spaces or tabs). 
Sequence consists of two allele variants in square brackets, separated with `/`, and flanking sequences at both sides. Length of flanking sequences should be sufficient to place the longest motif of a given collection (so it is advised to provide 25-30bp at each side) into all positions relative to a nucleotide substitution position. 

\exampleof{SNV list}
\noindent\texttt{%
rs10040172 gatttgccctgattgcagttactga[G/A]tggtacagacatcgtaataatctta
rs10116271 taaattctatgtggggaagaggtct[C/T]gtagaggcgatgattcttacattgc
rs10208293 cttcatacatttatgtccagtacct[A/G]tggaccctccttgtgaactcttctc
rs10431961 tggcggggctggtcaggcggcgtcg[C/T]cggtacgctctgagcggcagcgtgt
}
