\cmdname{SNPScan} uses a list of sequences with SNVs as input data.

The list of sequences with SNVs should be given in a single plain text file. 
Each sequence should be presented at a separate line using the following format:\\
\texttt{\inangles{SNV name} \inangles{left flank}[\inangles{variant 1}/\inangles{variant 2}]\inangles{right flank}}

SNV name shouldn't contain empty delimiters (spaces or tabs). 
Sequence consists of two allele variants in square brackets, separated with `/`, and flanking sequences at both sides. Length of flanking sequences should be sufficient to place the longest motif of a given collection (so it is advised to provide 25-30bp at each side) into all positions relative to a nucleotide substitution position. 

So, first two columns are SNV name and SNV sequence. Later columns (if present) are ignored, thus can contain any data.

\exampleof{SNV list}
\noindent\texttt{%
\# Text after "\#" doesn't matter\\
\# It's possible to include any number of comment lines into input\\
rs10040172 gattgcagttactga[G/A]tggtacagacatcgt\quad Anything\\
rs10116271 gtggggaagaggtct[C/T]gtagaggcgatgatt\quad can go\\
rs10208293 ttatgtccagtacct[A/G]tggaccctccttgtg\quad after first\\
rs10431961 ggtcaggcggcgtcg[C/T]cggtacgctctgagc\quad two columns
}

\vspace{1.5ex}
Note that lines starting with \texttt{\#} are considered as comments and thus ignored by \cmdname{SNPScan}.
